### PR TITLE
Saiyan v2

### PR DESCRIPTION
**Pull request by nsmith-**
@mcremone this is your PR but without the large nanoaod file.  I made a smaller version with just 10 events.

### CODE REVIEW

**Comment by nsmith-** [comment]
Note that the tests still fail, but it appears for 'real' reasons.

**Comment by mcremone** [comment]
Which failures did you get? Keep in mind that this works with specific versions of awkward and uproot, as you may remember:

pip install --user uproot==3.3.2
pip install --user uproot-methods==0.3.3
pip install --user awkward==0.7.1

**Comment by lgray** [comment]
@mcremone We cannot support those versions they have far more bugs in them compared to the modern releases of the software. Update your code.

**Comment by lgray** [comment]
@mcremone You can see how the test you wrote for your code failed if you click on 'details' in the one of the Travis CI runs. It saves the full logs.

**Comment by mcremone** [comment]
@nsmith- code cannot be updated. We need some fixes on the awkward/uproot side before moving to newer versions. Jim is working on those. For the moment I suggest we keep this PR open until we are able to move to newer awkward/uproot.

**Comment by mcremone** [comment]
https://github.com/LPC-DM/decaf/issues/19

**Comment by lgray** [comment]
Any movement here?

**Comment by lgray** [comment]
Rebased this PR to master.  Please conform to flake8 and fix package versioning issues.

**Comment by lgray** [comment]
```
Collecting typing; python_version < "3.5" fnal_column_analysis_tools/arrays/Builder.py:4:1: E302 expected 2 blank lines, found 1
fnal_column_analysis_tools/arrays/Builder.py:5:25: E231 missing whitespace after ','
fnal_column_analysis_tools/arrays/Builder.py:8:1: W293 blank line contains whitespace
fnal_column_analysis_tools/arrays/Builder.py:9:1: E302 expected 2 blank lines, found 1
fnal_column_analysis_tools/arrays/Builder.py:10:1: W293 blank line contains whitespace
fnal_column_analysis_tools/arrays/Builder.py:12:50: W291 trailing whitespace
fnal_column_analysis_tools/arrays/Builder.py:14:63: E225 missing whitespace around operator
fnal_column_analysis_tools/arrays/Builder.py:16:39: E225 missing whitespace around operator
fnal_column_analysis_tools/arrays/Builder.py:18:1: W293 blank line contains whitespace
fnal_column_analysis_tools/arrays/Builder.py:22:42: E225 missing whitespace around operator
fnal_column_analysis_tools/arrays/Builder.py:22:71: E227 missing whitespace around bitwise or shift operator
fnal_column_analysis_tools/arrays/Builder.py:22:97: E225 missing whitespace around operator
fnal_column_analysis_tools/arrays/Builder.py:22:126: E202 whitespace before ')'
fnal_column_analysis_tools/arrays/Builder.py:25:28: E225 missing whitespace around operator
fnal_column_analysis_tools/arrays/Builder.py:27:31: E225 missing whitespace around operator
fnal_column_analysis_tools/arrays/Builder.py:27:60: E227 missing whitespace around bitwise or shift operator
fnal_column_analysis_tools/arrays/Builder.py:27:76: E225 missing whitespace around operator
fnal_column_analysis_tools/arrays/Builder.py:27:105: E202 whitespace before ')'
fnal_column_analysis_tools/arrays/Builder.py:28:13: E101 indentation contains mixed spaces and tabs
fnal_column_analysis_tools/arrays/Builder.py:28:13: W191 indentation contains tabs
fnal_column_analysis_tools/arrays/Builder.py:28:25: E225 missing whitespace around operator
fnal_column_analysis_tools/arrays/Builder.py:30:28: E225 missing whitespace around operator
fnal_column_analysis_tools/arrays/Builder.py:32:1: W293 blank line contains whitespace
fnal_column_analysis_tools/arrays/Builder.py:39:29: E231 missing whitespace after ','
fnal_column_analysis_tools/arrays/Builder.py:40:72: E231 missing whitespace after ','
fnal_column_analysis_tools/arrays/Builder.py:40:75: E231 missing whitespace after ','
fnal_column_analysis_tools/arrays/Builder.py:40:80: E231 missing whitespace after ','
fnal_column_analysis_tools/arrays/Builder.py:41:72: E127 continuation line over-indented for visual indent
fnal_column_analysis_tools/arrays/Builder.py:41:76: E231 missing whitespace after ','
fnal_column_analysis_tools/arrays/Builder.py:41:79: E231 missing whitespace after ','
fnal_column_analysis_tools/arrays/Builder.py:41:84: E231 missing whitespace after ','
fnal_column_analysis_tools/arrays/Builder.py:44:75: E231 missing whitespace after ','
fnal_column_analysis_tools/arrays/Builder.py:45:76: E231 missing whitespace after ','
fnal_column_analysis_tools/arrays/Builder.py:52:74: E231 missing whitespace after ','
fnal_column_analysis_tools/arrays/Builder.py:53:75: E231 missing whitespace after ','
fnal_column_analysis_tools/arrays/Builder.py:60:70: E231 missing whitespace after ','
fnal_column_analysis_tools/arrays/Builder.py:61:70: E231 missing whitespace after ','
fnal_column_analysis_tools/arrays/Builder.py:68:77: E231 missing whitespace after ','
fnal_column_analysis_tools/arrays/Builder.py:69:77: E231 missing whitespace after ','
fnal_column_analysis_tools/arrays/Builder.py:76:75: E231 missing whitespace after ','
fnal_column_analysis_tools/arrays/Builder.py:77:75: E231 missing whitespace after ','
fnal_column_analysis_tools/arrays/Builder.py:84:74: E231 missing whitespace after ','
fnal_column_analysis_tools/arrays/Builder.py:85:76: E231 missing whitespace after ','
fnal_column_analysis_tools/arrays/Builder.py:92:68: E231 missing whitespace after ','
fnal_column_analysis_tools/arrays/Builder.py:114:10: E261 at least two spaces before inline comment
```

**Comment by lgray** [comment]
@mcremone As discussed on skype. Got this working with all the latest awkward and uproot. You should try running your analysis with this branch of fnal_column_analysis_tools and all the latest dependencies. Then you should move to using this package specifically.

File: tests/test_arrays.py [hunk 1]
```diff
@@ -131,4 +126,7 @@ def test_object_arrays():
     #print(met[zeroL.sum()&skinny.sum()])
     #print(j_clean[j_clean.pt.argmax()].pt)
     #print(met.sum()+mu[mu.pt.argmax()].sum())
-    print(dimu[leading_dimu].sum())
+    if dimu.content.size > 0:
+        print(dimu[dimu.pt.arxmax()].sum())
```
Comment (nsmith-) [inline tests/test_arrays.py]: `arxmax`? Is there still bugs with zero-length arrays?

**Comment by lgray** [comment]
Yeah, that's even weirder!

**Comment by areinsvo** [comment]
I tested this with our analysis code and up-to-date versions of awkward, uproot, and uproot_methods. Everything worked fine. From my point of view it can be merged.

**Comment by lgray** [comment]
I'll make a release tomorrow.

**Pull request merged**